src/api/spec/controllers/webui/main_controller_spec.rb

```diff
@@ -70,4 +70,57 @@
       end
     end
   end
+
+  describe "GET #sitemap" do
+    render_views
+
+    before do
+      get :sitemap
+      @paths = Hash.from_xml(response.body)['sitemapindex']['sitemap'].map do |url|
+          uri = URI.parse(url['loc'])
+          "#{uri.path}?#{uri.query}"
+      end
+    end
+
+    it { expect(@paths).to include('/main/sitemap_projects?') }
+    it "have all category's urls" do
+      ('a'..'z').to_a.concat(('A'..'Z').to_a).each do |letter|
+        expect(@paths).to include("/main/sitemap_packages/show?category=home%3A#{letter}")
+      end
+    end
+    it { expect(@paths).to include("/main/sitemap_packages/show?category=opensuse") }
+  end
+
+  describe "GET #sitemap_projects" do
+    render_views
+
+    before do
+      create(:confirmed_user)
+      5.times { create(:project) }
+      get :sitemap_projects
+      @project_paths = Hash.from_xml(response.body)['urlset']['url'].map { |url| URI.parse(url['loc']).path }
+    end
+
+    it "have all project's urls" do
+      Project.all.pluck(:name).each do |project_name|
+          expect(@project_paths).to include("/project/show/#{project_name}")
```

Comment:
That is testing that all projects urls are in the xml returned by the action... isn't it?